### ISSUE TITLE
JsonPath template query in config causing startup failure

### ISSUE THREAD

**Issue by movinfinex** [outofcoffee/imposter#631]
Simply including a JsonPath template query in a config (even in a comment) causes Imposter to fail at startup:

E.g.:
```
---
plugin: rest
# ${context.request.body:$.foo}
```

```
[imposter] | 05:07:19 DEBUG i.g.i.c.u.ConfigUtil - Loading configuration file: /opt/imposter/config/test-config.yaml
[imposter] | 05:07:19 TRACE i.g.i.e.u.ExpressionUtil - Evaluating expression: context.request.body:$.foo
[imposter] | 05:07:19 TRACE i.g.i.e.u.ExpressionUtil - Unsupported expression: context.request.body:$.foo, evaluators: [env, system]
[imposter] | 05:07:19 TRACE i.g.i.e.u.ExpressionUtil - context.request.body:$.foo=\${context.request.body:$.foo}
[imposter] | 05:07:19 ERROR i.v.c.i.l.c.VertxIsolatedDeployer - Failed in deploying verticle
[imposter] | java.lang.RuntimeException: Failed to read plugin config: ConfigReference(file=/opt/imposter/config/test-config.yaml, configRoot=/opt/imposter/config)
[imposter] |    at io.gatehill.imposter.config.util.ConfigUtil.readPluginConfigs(ConfigUtil.kt:206) ~[imposter-config-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.Imposter.processConfiguration(Imposter.kt:158) ~[imposter-engine-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.Imposter.access$processConfiguration(Imposter.kt:88) ~[imposter-engine-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.Imposter$start$1.invokeSuspend(Imposter.kt:119) ~[imposter-engine-4.0.3.jar:?]
[imposter] |    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.9.10.jar:1.9.10-release-459]
[imposter] |    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108) ~[kotlinx-coroutines-core-jvm-1.7.3.jar:?]
[imposter] |    at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584) ~[kotlinx-coroutines-core-jvm-1.7.3.jar:?]
[imposter] |    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793) ~[kotlinx-coroutines-core-jvm-1.7.3.jar:?]
[imposter] |    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697) ~[kotlinx-coroutines-core-jvm-1.7.3.jar:?]
[imposter] |    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684) ~[kotlinx-coroutines-core-jvm-1.7.3.jar:?]
[imposter] | Caused by: java.lang.RuntimeException: Error evaluating expression: context.request.body:$.foo
[imposter] |    at io.gatehill.imposter.expression.util.ExpressionUtil.eval(ExpressionUtil.kt:95) ~[imposter-expression-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.expression.util.ExpressionUtil.eval$default(ExpressionUtil.kt:77) ~[imposter-expression-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.config.util.ConfigUtil.readPluginConfig$imposter_config(ConfigUtil.kt:266) ~[imposter-config-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.config.util.ConfigUtil.readPluginConfigs(ConfigUtil.kt:198) ~[imposter-config-4.0.3.jar:?]
[imposter] |    ... 9 more
[imposter] | Caused by: java.lang.IllegalArgumentException: Illegal group reference
[imposter] |    at java.util.regex.Matcher.appendExpandedReplacement(Unknown Source) ~[?:?]
[imposter] |    at java.util.regex.Matcher.appendReplacement(Unknown Source) ~[?:?]
[imposter] |    at io.gatehill.imposter.expression.util.ExpressionUtil.eval(ExpressionUtil.kt:93) ~[imposter-expression-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.expression.util.ExpressionUtil.eval$default(ExpressionUtil.kt:77) ~[imposter-expression-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.config.util.ConfigUtil.readPluginConfig$imposter_config(ConfigUtil.kt:266) ~[imposter-config-4.0.3.jar:?]
[imposter] |    at io.gatehill.imposter.config.util.ConfigUtil.readPluginConfigs(ConfigUtil.kt:198) ~[imposter-config-4.0.3.jar:?]
[imposter] |    ... 9 more
```

Escaping the second dollar sign (`${context.request.body:\$.foo}`) allows Imposter to start up, and the expression actually works if used in a capture for example.

Is this the expected behaviour?  I can make a PR to update the docs at https://docs.imposter.sh/template_queries/

**Comment by outofcoffee** [comment]
Hi @movinfinex, thank you for raising this.

Unsupported expressions should be ignored when parsing config - this will need to be fixed in a new release.

**Comment by outofcoffee** [comment]
Fixed in [release 4.0.4](https://github.com/outofcoffee/imposter/releases).

**Comment by movinfinex** [comment]
Thanks, I've pulled the update.  The template query works using the normal syntax in 4.04 (though the undocumented escaped version breaks).
